ops/op2/dispatch_fast.rs

```diff
@@ -323,36 +323,30 @@ pub(crate) fn get_fast_signature(
   }))
 }
 
-/// Sheds the error in a `Result<T, E>` as an early return, leaving just the `T` and requesting
-/// that v8 re-call the slow function to throw the error.
-pub(crate) fn generate_fast_result_early_exit(
+pub(crate) fn return_value_result(
   generator_state: &mut GeneratorState,
 ) -> TokenStream {
-  generator_state.needs_fast_api_callback_options = true;
   generator_state.needs_fast_opctx = true;
-  gs_quote!(generator_state(fast_api_callback_options, opctx, result) => {
+
+  let maybe_scope = if generator_state.needs_scope {
+    quote!()
+  } else {
+    with_scope(generator_state)
+  };
+
+  gs_quote!(generator_state(opctx, scope, result) => {
     let #result = match #result {
       Ok(#result) => #result,
       Err(err) => {
-        let err = err.into();
-
-        // FASTCALL FALLBACK: This is where we set the errors for the slow-call error pickup path. There
-        // is no code running between this and the other FASTCALL FALLBACK comment, except some V8 code
-        // required to perform the fallback process. This is why the below call is safe.
-
-        // The reason we need to do this is because V8 does not allow exceptions to be thrown from the
-        // fast call. Instead, you are required to set the fallback flag, which indicates to V8 that it
-        // should re-call the slow version of the function. Technically the slow call should perform the
-        // same operation and then throw the same error (because it should be idempotent), but in our
-        // case we stash the error and pick it up on the slow path before doing any work.
-
-        // TODO(mmastrac): We should allow an #[op] flag to re-perform slow calls without the error path when
-        // the method is performance sensitive.
+        #maybe_scope
 
-        // SAFETY: We guarantee that OpCtx has no mutable references once ops are live and being called,
-        // allowing us to perform this one little bit of mutable magic.
-        unsafe { #opctx.unsafely_set_last_error_for_ops_only(err); }
-        #fast_api_callback_options.fallback = true;
+        let err = err.into();
+        let exception = deno_core::error::to_v8_error(
```

Comment:
This works now because we can now alloc in fast calls?

Reply:
yes